quickstarts/microsoft.web/function-premium-vnet-integration/main.bicep

```diff
@@ -0,0 +1,149 @@
+param appInsightsLocation string
+
+@description('The language worker runtime to load in the function app.')
+@allowed([
+  'node'
+  'dotnet'
+  'java'
+])
+param runtime string = 'node'
+
+@description('The name of the function app that you wish to create.')
+param appName string = 'fnapp${uniqueString(resourceGroup().id)}'
+
+@allowed([
+  'Standard_LRS'
+  'Standard_GRS'
+  'Standard_RAGRS'
+])
+@description('Storage Account type')
+param storageAccountType string = 'Standard_LRS'
+
+@description('The name of the virtual network to be created.')
+param vnetName string
+
+@description('The name of the subnet to be created within the virtual network.')
+param subnetName string
+
+var vnetAddressPrefix = '10.0.0.0/16'
+var subnetAddressPrefix = '10.0.0.0/24'
+var functionAppName = appName
+var hostingPlanName = appName
+var storageAccountName = '${uniqueString(resourceGroup().id)}azfunctions'
+var functionWorkerRuntime = runtime
+
+resource virtualNetwork 'Microsoft.Network/virtualNetworks@2020-06-01' = {
+  name: vnetName
+  location: location
+  properties: {
+    addressSpace: {
+      addressPrefixes: [
+        vnetAddressPrefix
+      ]
+    }
+    subnets: [
+      {
+        name: subnetName
+        properties: {
+          addressPrefix: subnetAddressPrefix
+          delegations: [
+            {
+              name: 'delegation'
+              properties: {
+                serviceName: 'Microsoft.Web/serverFarms'
+              }
+            }
+          ]
+        }
+      }
+    ]
+  }
+}
+
+resource storageAccount 'Microsoft.Storage/storageAccounts@2021-04-01' = {
+  name: storageAccountName
+  location: location
+  sku: {
+    name: storageAccountType
+  }
+  kind: 'StorageV2'
+}
+
+resource appInsights 'Microsoft.Insights/components@2018-05-01-preview' = {
+  name: functionAppName
+  location: appInsightsLocation
+  kind: 'web'
+  properties: {
+    Application_Type: 'web'
+  }
+}
+
+resource serverFarm 'Microsoft.Web/serverfarms@2020-06-01' = {
+  name: hostingPlanName
+  location: location
+  sku: {
+    name: 'EP1'
+    tier: 'ElasticPremium'
+  }
+  kind: 'elastic'
+  properties: {
+    maximumElasticWorkerCount: 20
+  }
+}
+
+resource function 'Microsoft.Web/sites@2020-06-01' = {
+  name: functionAppName
+  location: location
+  kind: 'functionapp'
+  properties: {
+    serverFarmId: serverFarm.id
+    siteConfig: {
+      appSettings: [
+        {
+          name: 'APPINSIGHTS_INSTRUMENTATIONKEY'
+          value: appInsights.properties.InstrumentationKey
+        }
+        {
+          name: 'APPLICATIONINSIGHTS_CONNECTION_STRING'
+          value: 'InstrumentationKey=${appInsights.properties.InstrumentationKey}'
+        }
+        {
+          name: 'AzureWebJobsStorage'
+          value: 'DefaultEndpointsProtocol=https;AccountName=${storageAccount.name};EndpointSuffix= ${environment().suffixes.storage};AccountKey=${listkeys(storageAccount.id, '2021-04-01').keys[0].value}'
```

Comment:
I think should we follow the same pattern here (as in the others) and use ```storageAccount.apiVersion``` for the api version